src/TransformGraphVisitors.hpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * TransformGraphExceptions.hpp
+ *
+ *  Created on: Sep 22, 2015
+ *      Author: jhidalgo
+ */
+
+#pragma once
+#include "TransformGraphExceptions.hpp"
+#include <boost/shared_ptr.hpp>
+#include <boost/graph/breadth_first_search.hpp>
+
+#include <deque>
+#include <unordered_map>
+
+namespace envire { namespace core
+{
+
+    class Empty{};
+
+    template < typename Vertex>
+    class TransformGraphBFSVisitor: public boost::default_bfs_visitor
+    {
+    public:
+        template <typename Graph>
+        TransformGraphBFSVisitor(const Vertex &v,const Graph &g):vertex_to_search(v)
+        {
+            parent.reset(new std::unordered_map<Vertex,Vertex>());
+            tree.reset(new std::deque<Vertex>());
+        }
+
+        template <typename Graph>
+        void discover_vertex(Vertex v, const Graph &g)
+        {
+            if (vertex_to_search == v)
+            {
+                Vertex parent_vertex = v;
+
+                /** Get its direct parent **/
+                while (parent_vertex != Graph::null_vertex())
+                {
+                    tree->push_front(parent_vertex);
+                    try
+                    {
+                        parent_vertex = parent->at(parent_vertex);
+                    }catch (const std::out_of_range& oor)
+                    {
+                        parent_vertex = Graph::null_vertex();
+                    }
+                }
+
+                throw FoundFrameException(g[v].frame.name);
```

Comment:
Is this the normal way to return a value from boost visitors?
It seems a bit strange to use an exception.